swagger-models/src/test/groovy/com/mangofactory/swagger/models/property/bean/BeanModelPropertyProviderSpec.groovy

```diff
@@ -0,0 +1,36 @@
+package com.mangofactory.swagger.models.property.bean
+
+import com.fasterxml.classmate.ResolvedType
+import com.fasterxml.classmate.TypeResolver
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.mangofactory.swagger.mixins.ModelPropertyLookupSupport
+import com.mangofactory.swagger.mixins.TypesForTestingSupport
+import com.mangofactory.swagger.models.ObjectMapperBeanPropertyNamingStrategy
+import com.mangofactory.swagger.models.alternates.AlternateTypeProvider
+import spock.lang.Specification
+
+
+@Mixin([TypesForTestingSupport, ModelPropertyLookupSupport])
+class BeanModelPropertyProviderSpec extends Specification {
+
+
+  def "Respect property ordering" () {
+
+    given:
+      Class typeToTest = complexType()
+      def typeResolver = new TypeResolver()
+      ResolvedType resolvedType = typeResolver.resolve(typeToTest)
+      ObjectMapper mapper = new ObjectMapper()
+
+      def beanModelPropertyProvider = new BeanModelPropertyProvider(new AccessorsProvider(typeResolver), typeResolver,
+            new AlternateTypeProvider(), new ObjectMapperBeanPropertyNamingStrategy(mapper))
+      beanModelPropertyProvider.objectMapper = mapper
+      def properties = beanModelPropertyProvider.propertiesForSerialization(resolvedType)
```

Comment:
speficifically, does the following also work, for weird edge cases in the way jackson allows you to configure beans! :)

```java
def properties = beanModelPropertyProvider.propertiesForDeserialization(resolvedType)
```